internal/auth0/user/resource_roles_test.go

```diff
@@ -0,0 +1,92 @@
+package user_test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+
+	"github.com/auth0/terraform-provider-auth0/internal/acctest"
+)
+
+const givenAUserAndRoles = `
+resource auth0_role owner {
+	name = "owner"
+	description = "Owner"
+}
+
+resource auth0_role admin {
+	depends_on = [auth0_role.owner]
+
+	name = "admin"
+	description = "Administrator"
+}
+
+resource auth0_user user {
+	depends_on = [auth0_role.owner, auth0_role.admin]
+
+	connection_name = "Username-Password-Authentication"
+	email = "{{.testName}}@acceptance.test.com"
+	password = "passpass$12$12"
+
+	lifecycle {
+		ignore_changes = [roles]
+	}
+}
+`
+
+const createUserRolesWithNoRoles = givenAUserAndRoles + `
+resource auth0_user_roles user_roles {
+	depends_on = [ auth0_user.user ]
+
+	user_id = auth0_user.user.id
+	roles = []
+}
+`
+
+const updateUserRolesWithRoles = givenAUserAndRoles + `
+resource auth0_user_roles user_roles {
+	depends_on = [ auth0_user.user ]
+
+	user_id = auth0_user.user.id
+	roles = [
+		auth0_role.owner.id,
+		auth0_role.admin.id
+	]
+}
+`
+
+const deleteUserRolesResource = `
+resource auth0_user user {
+	connection_name = "Username-Password-Authentication"
+	email = "{{.testName}}@acceptance.test.com"
+	password = "passpass$12$12"
+}
+`
+
+func TestAccUserRoles(t *testing.T) {
+	testName := strings.ToLower(t.Name())
+
+	acctest.Test(t, resource.TestCase{
+		Steps: []resource.TestStep{
+			{
+				Config: acctest.ParseTestName(createUserRolesWithNoRoles, testName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("auth0_user_roles.user_roles", "roles.#", "0"),
+				),
+			},
+			{
+				Config: acctest.ParseTestName(updateUserRolesWithRoles, testName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("auth0_user_roles.user_roles", "roles.#", "2"),
+				),
+			},
+			{
```

Comment:
I think it would be nice to have an intermediate step between two roles and zero roles where a single role is removed from the set.